src/v/pandaproxy/schema_registry/avro.cc

```diff
@@ -307,17 +327,19 @@ result<void> sanitize(json::Value::Object& o, sanitize_context& ctx) {
                   it->value.GetString(), it->value.GetStringLength()};
                 if (existing_namespace != new_namespace) {
                     it->value.SetString(
-                      new_namespace.data(),
-                      new_namespace.length(),
+                      new_namespace->data(),
+                      new_namespace->length(),
                       ctx.alloc);
                 }
             } else {
                 o.AddMember(
                   json::Value("namespace"),
                   json::Value(
-                    new_namespace.data(), new_namespace.length(), ctx.alloc),
+                    new_namespace->data(), new_namespace->length(), ctx.alloc),
                   ctx.alloc);
             }
+        } else {
+            o.RemoveMember("namespace");
```

Comment:
I believe control could get to this branch in the case where a field as a name and the **null** namespace. In such a case, would this call to `o.RemoveMember("namespace")` cause a crash?

For example, what if someone posts the following:
```
{
  "type": "record",
  "name": "Example",
  "doc": "A simple name (attribute) and no namespace attribute: use the null namespace (\"\"); the fullname is 'Example'.",
  "fields": [
    {
      "name": "inheritNull",
      "type": {
        "type": "enum",
        "name": "Simple",
        "doc": "A simple name (attribute) and no namespace attribute: inherit the null namespace of the enclosing type 'Example'. The fullname is 'Simple'.",
        "symbols": ["a", "b"]
      }
    }
  ]
}
```

Reply:
Good spot, `o.RemoveMember("namespace")` will return `false`.

Reply:
Confirmed that this will return false if the member doesn't exist:
From RapidJson https://github.com/Tencent/rapidjson/blob/master/include/rapidjson/document.h#L1527-L1538
```cpp
    //! Remove a member in object by its name.
    /*! \param name Name of member to be removed.
        \return Whether the member existed.
        \note This function may reorder the object members. Use \ref
            EraseMember(ConstMemberIterator) if you need to preserve the
            relative order of the remaining members.
        \note Linear time complexity.
    */
    bool RemoveMember(const Ch* name) {
        GenericValue n(StringRef(name));
        return RemoveMember(n);
    }
```

Reply:
For posterity I will note that sanitization requires a correct order of members in the object, but that sorting occurs after this call.